tests/char_stamina_test.cpp

```diff
@@ -97,8 +97,8 @@ static void burden_player( player &dummy, float burden_proportion )
         dummy.i_add( pile );
     }
 
-    // Ensure we are carrying the expected amount of weight
-    REQUIRE( dummy.weight_carried() == capacity * burden_proportion );
+    // Ensure we are carrying the expected weight (accurate to 1 g)
+    REQUIRE( dummy.weight_carried() / 1000 == capacity * burden_proportion / 1000 );
```

Comment:
Maybe it should instead compare with `units` (from line 95) on the right side.

Reply:
Good idea, thanks - and I believe `to_grams( dummy.weight_carried() )` will work better for the left hand side.